### ISSUE TITLE
Prevent setting the date of a scheduled update to a time in the past

### ISSUE THREAD

**Issue by coreymckrill** [WordPress/revisions-extended#56]
If you edit a scheduled update, and change the date to sometime in the past, the button changes to "Publish", but clicking it doesn't actually publish anything (which is good, since revisions should never be published themselves). Ideally, I think, the editor would just show an error message if you tried to set a date in the past.

**Comment by bph** [comment]
It does that now, see #65

**Comment by StevenDufresne** [comment]
This issue different from #65. It's referencing the time picker when you are specifically editing an **update**. 

#65 occurs when editing a specific post.

**Comment by StevenDufresne** [comment]
Do we want to use the default Gutenberg messaging system or try something in-line? 

Also, I looked into controlling the minimum date but the Gutenberg control, which wraps a third party one that does support the feature, doesn't expose the `minDate` property.

**Screenshot**

![](https://d.pr/i/O2tGZq.png)

**Comment by coreymckrill** [comment]
Is this screenshot the default GB messaging, or something custom? Default would be my first preference. The screenshot looks fine.

> the Gutenberg control, which wraps a third party one that does support the feature, doesn't expose the minDate property.

D'oh. That might be worth filing an issue in the GB repo.

**Comment by StevenDufresne** [comment]
> Is this screenshot the default GB messaging, or something custom? Default would be my first preference. The screenshot looks fine.

No, Gutenberg doesn't have a messaging system within the sidebar (75% sure :)). The default messaging is the one that appears at the top of the document that we have already used as the main update editing landmark.

**Comment by coreymckrill** [comment]
Here's a variation using WP Admin's `notice` and `notice-error` classes for the message container:

![select-future](https://user-images.githubusercontent.com/916023/109231462-802c4600-777b-11eb-9be6-a4a67a705306.jpg)

**Comment by coreymckrill** [comment]
#94 partially addresses this issue because the datepicker in our custom sidebar connects to the "create update" button, and the button becomes disabled if a date in the past is picked. However, when editing a revision, it's still possible to pick a past date and then click the save button (which changes from "Update" to "Publish")

**Comment by StevenDufresne** [comment]
**Noting**: I looked into adding a PR for Gutenberg to accept a `minDate` since the underlying datepicker library does support it. However, Gutenberg imports a very specific version of the picker from that library that does not support it (https://github.com/airbnb/react-dates/blob/7a93b097af2e4f4ca8f8e6a329b213f49d70bd98/src/components/DayPickerSingleDateController.jsx#L37). 

For now, we'll need to manage this internally.
